src/Database/ClientState.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4: */
+/**
+ * Client State class for managing client_states table
+ *
+ * @category Database
+ * @package  Helioviewer
+ * @author   Kasim Necdet Percinel <kasim.n.percinel@nasa.gov>
+ * @license  http://www.mozilla.org/MPL/MPL-1.1.html Mozilla Public License 1.1
+ * @link     https://github.com/Helioviewer-Project/
+ */
+
+require_once HV_ROOT_DIR.'/../src/Database/DbConnection.php';
+
+class ClientState extends Database_DbConnection 
+{
+    /**
+     * This function creates or updates the state in database.
+     *
+     * @return string
+     */
+    public function upsert(array $state): string 
+    {
+        $state_json = json_encode($state); 
+        $state_key = md5($state_json);
+
+        $create_sql = "REPLACE INTO client_states(id, state) VALUES ('%s','%s')";
+        $create_state_sql = sprintf($create_sql, $state_key, $this->link->real_escape_string($state_json));
+
+        // intentionally let exception thrown
+        $result = $this->query($create_state_sql);
```

Comment:
+1
Thanks for not copying the bad practice of returning false. It has caused me so many headaches.